cmd/dagger/shell.go

```diff
@@ -285,9 +287,23 @@ func (h *shellCallHandler) run(ctx context.Context, reader io.Reader, name strin
 	}
 
 	resp, err := h.Result(ctx, h.stdoutBuf, true, handleObjectLeaf)
-	if err != nil || resp == nil {
+	if err != nil {
+		exitCode := 1
+		var ex *dagger.ExecError
+		if errors.As(err, &ex) {
+			if ex.Stderr != "" && (h.repl || !h.tty) {
+				return fmt.Errorf("%w\n%s", err, ex.Stderr)
```

Comment:
May also be worth printing stdout as well?

Reply:
Yeah, I went back and forth on that, but just added it in just in case.